src/routes/subscriptions.js

```diff
@@ -1,128 +1,31 @@
-
 const express = require('express');
 const router = express.Router();
-const axios = require('axios');
-const db = require('../storage/DataAccess');
-const { startReportingWorkflow } = require('../utils/reporting_workflow');
-const { getServerById } = require('../storage/servers');
-const {
-  refreshKnowledgeArtifact,
-  getEndpointId,
-  subscriptionsFromPlanDef,
-  postSubscriptionsToEHR,
-  getBaseUrlFromFullUrl
-} = require('../utils/fhir');
-const { getAccessToken } = require('../utils/client');
-const { PLANDEFINITIONS, ENDPOINTS } = require('../storage/collections');
-const { default: base64url } = require('base64url');
-const debug = require('../storage/logs').debug('medmorph-backend:subscriptions');
+const notificationService = require('../services/subscriptions');
 
 /**
  * Notification from KA Repo to refresh artifacts. The id param is the
  * server id from the database
  */
-router.post('/ka/:id', (req, res) => {
-  const id = req.params.id;
-  const server = getServerById(id);
-  if (server) {
-    refreshKnowledgeArtifact(server);
-    debug(`Recieved notification to refresh knowledge artifacts for server with id ${id}`);
-    res.sendStatus(StatusCodes.OK);
-  } else res.sendStatus(StatusCodes.NOT_FOUND);
-});
+router.post('/ka/:id', notificationService.knowledgeArtifact);
 
 /**
  * Notification from KA Repo to refresh artifacts. The id param is the
  * server id from the database and resourceId is the PlanDefinition id.
  * The body contains the triggering PlanDefinition
  */
-router.put('/ka/:id/:resource/:resourceId', async (req, res) => {
-  const id = req.params.id;
-  const planDefinition = req.body;
-  const endpointId = getEndpointId(planDefinition);
-  const server = getServerById(id);
-
-  if (!server) {
-    res.sendStatus(StatusCodes.NOT_FOUND);
-    return;
-  }
-
-  const planDefinitionFullUrl = `${server.endpoint}/PlanDefinition/${planDefinition.id}`;
-  db.upsert(
-    PLANDEFINITIONS,
-    { fullUrl: planDefinitionFullUrl, ...planDefinition },
-    r => r.fullUrl === planDefinitionFullUrl
-  );
-  debug(`Fetched ${server.endpoint}/PlanDefinition/${planDefinition.id}`);
-
-  const token = await getAccessToken(server.endpoint);
-  const headers = { Authorization: `Bearer ${token}` };
-  axios.get(`${server.endpoint}/Endpoint/${endpointId}`, { headers: headers }).then(response => {
-    if (response.data) {
-      const endpoint = response.data;
-      const endpointFullUrl = `${server.endpoint}/Endpoint/${endpoint.id}`;
-      db.upsert(
-        ENDPOINTS,
-        { fullUrl: endpointFullUrl, ...endpoint },
-        r => r.fullUrl === endpointFullUrl
-      );
-      debug(`Fetched ${endpointFullUrl}`);
-    }
-  });
-
-  const subscriptions = subscriptionsFromPlanDef(planDefinition, server.endpoint);
-  postSubscriptionsToEHR(subscriptions);
-
-  res.sendStatus(StatusCodes.OK);
-});
+router.put('/ka/:id/:resource/:resourceId', notificationService.knowledgeArtifact);
```

Comment:
Do we still need this endpoint? It doesn't look like `knowledgeArtifact` function uses the `resourceId` param.

Reply:
So `resource` and `resourceId` are not used but I left this endpoint in... it's unclear at this point whether the backport will POST to {notification url} and/or PUT to {notification url}/Bundle/{bundleid}. So I decided to keep both endpoints for now. However changing this to `router.put('ka/:id, notificationService.knowledgeArtifact);` will give a 404 error for PUT to {notification url}/Bundle/{bundleid}. So even though the value isn't used its still needed for the endpoint

Reply:
Oh ok that makes sense, thanks.